webextension/scripts/login.js

```diff
@@ -0,0 +1,64 @@
+$('#sign_up').click(signUp)
+$('#forgot_password').click(forgotPassword)
+$('#log_in').click(doLogin)
+$('#logout').click(doLogout)
+
+function signUp(){
+    openByWindowSetting('https://archive.org/account/signup')
+}
+
+function forgotPassword(){
+    openByWindowSetting('https://archive.org/account/forgot-password')
+}
+
+function doLogin(e){
+    e.preventDefault()
+    $('#error_message').hide()
+    let email = $('#email-address').val()
+    let password = $('#password').val()
+    if(email.length === 0){
+        $('#error_message').show().text('Please type an email')
+        return
+    }
+    if(password.length === 0){
+        $('#error_message').show().text('Please type a password')
+        return
+    }
+    $('#log_in').val('Please Wait...')
+    const data = new URLSearchParams()
+    data.append('username', email)
+    data.append('password',password)
+    data.append('action',"login")
+    const loginPromise = new Promise((resolve, reject) => {
+        setTimeout(() => {
+          reject(new Error('timeout'))
+        }, 30000)
+        fetch('https://archive.org/account/login',
+          {
+            method: 'POST',
+            body: data,
+            headers: {
+              'Accept': 'application/json',
+              'Content-Type': 'application/x-www-form-urlencoded'
+            }
+          })
+        .then(resolve, reject)
+      })
+    loginPromise
+      .then(response => response.json())
+      .then((res)=>{
+          if(res.status=='bad_login'){
+            $('#error_message').show().text('Incorrect Email or Password')
+            $('#log_in').val('Login')
+          }else{
+            $('#logout').show()
+          }
+      })
+      .catch(e=> console.log(e))
+}
+
+function doLogout(){
+    chrome.cookies.remove({url:'https://archive.org',name:'logged-in-sig'},()=>{
```

Comment:
Got the error `Uncaught TypeError: Cannot read property 'remove' of undefined` in the popup's console. It's probably due to the cookie being stored in a different process than the popup? I'm not sure, but maybe try moving this code to `background.js` then sending a message to call it (since can't call it directly in background process.) Or maybe it can be solved similar to `archive.js` which you can see it under "web_accessible_resources" in `manifest.json` - since that script has access to the website the user is viewing, you'd also need access in order to clear the cookie. I hope this makes sense!